### ISSUE TITLE
Secret table option breaks admin API requests

### ISSUE THREAD

**Issue by ethagnawl** [piccolo-orm/piccolo_admin#162]
Given a Piccolo table which looks like the following:

```
class Things(Table):
    id = UUID(primary_key=True)
    value = Text(index=True, null=False, required=True, secret=True)
```

... and a basic Piccolo admin which is configured using the `tables` kwarg:
```
...
create_admin(tables=APP_CONFIG.table_classes)
...
```

The dashboard's API request for all Thing records results in a 500 response and an error message which states: `Problem fetching thing rows.`

Removing the `secret` kwarg from the table definition solves the problem.

I don't see anything beyond the 500 reported in my application's logs but that may be a result of bogus configuration on my part. I'd be curios to know if there are any config settings which would increase log output or allow me to pass in a logger. I grepped through the admin panel's code base but didn't see anything relevant. Also, the only logging I saw being done was a few print statements. Again, I could very well have missed something(s).

I also suspect that using a `TableConfig` for this and excluding these fields when creating the admin instance would solve this problem but I haven't verified that yet.

Anyways, thanks for creating and supporting such a useful project! I was able to get an admin instance stood up with very little time and effort and look forward to molding it to my needs!

**Comment by sinisaos** [comment]
@ethagnawl As far as I understand, the point of the `Secret` column is that it does not display content in the UI because there may be some sensitive data, so the result in the rows for that Secret column is `null`. When it comes to PR, @dantownsend  will be a lot more competent than me.

**Comment by dantownsend** [comment]
Yeah, @sinisaos is right - having `required=True` and `secret=True` is problematic for the admin.

Having some kind of warning is a good idea. Perhaps here:

https://github.com/piccolo-orm/piccolo_admin/blob/bb2aa3955e55186bb4f1097b950b175b94040c86/piccolo_admin/endpoints.py#L250 

@ethagnawl A PR would be very welcome if you're happy to.

**Comment by ethagnawl** [comment]
@dantownsend @sinisaos See #164.

**Comment by ethagnawl** [comment]
Closed via #164.
